### ISSUE TITLE
SSL CERTIFICATE_VERIFY_FAILED when connecting to RHOCS registry backend

### ISSUE THREAD

**Issue by gorantornqvist** [quay/quay-operator#137]
I configured my QuayEcosystem for RH OCS 4.2 with:
```
spec:
  registryBackends:
      - name: rhocs
        rhocs:
          accessKey: *************
          secretKey: *************
          bucketName: quay-bucket-ec783efa-33d9-492a-a8b0-3fc42967ddb0
          hostname: s3.openshift-storage.svc
          secure: true
          port: 443
```
Also added openshift service ca cert to the config:
```
spec:
  configFiles:
      - secretName: openshift-service-ca
        files:
          - key: service-ca.crt
            type: extraCaCert
```
The cert gets injected to the quay-config pod and I can do:
```
sh-4.2# curl --cacert /conf/stack/extra_ca_certs/service-ca.crt https://s3.openshift-storage.svc
<?xml version="1.0" ...
```
(no SSL errors)

However I get this error for some reason:
```
2020-01-29 12:56:28,823 [64] [ERROR] [util.config.validator] Validation exception
Traceback (most recent call last):
  File "/quay-registry/util/config/validator.py", line 79, in validate_service_for_config
    VALIDATORS[service](validator_context)
  File "/quay-registry/util/config/validators/validate_storage.py", line 42, in validate
    "Invalid storage configuration: %s: %s" % (name, msg)
ConfigValidationException: Invalid storage configuration: rhocs: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)
gunicorn-config stdout | 2020-01-29 12:56:28,823 [64] [ERROR] [util.config.validator] Validation exception
Traceback (most recent call last):
  File "/quay-registry/util/config/validator.py", line 79, in validate_service_for_config
    VALIDATORS[service](validator_context)
  File "/quay-registry/util/config/validators/validate_storage.py", line 42, in validate
    "Invalid storage configuration: %s: %s" % (name, msg)
ConfigValidationException: Invalid storage configuration: rhocs: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)
10.254.7.68 () - quayconfig [29/Jan/2020:12:56:28 +0000] "POST /api/v1/superuser/config/validate/registry-storage HTTP/1.1" 200 141 "-" "Go-http-client/1.1" (62.204 2684 62.204)
```

quay-operator:v1.0.1

**Comment by sabre1041** [comment]
@gorantornqvist this has been fixed with the latest release. can you verify?

**Comment by simonkrenger** [comment]
@sabre1041 We are also observing this issue with version 1.0.2 (see also [RHBZ#1812141](https://bugzilla.redhat.com/show_bug.cgi?id=1812141))

**Comment by kurtismullins** [comment]
@simonkrenger or @gorantornqvist 

Can you provide the steps to reproduce a similar certificate that you generated? I want to try to reproduce this on my end.

**Comment by alecmerdler** [comment]
Resolved in #239
